plugins/multilanguage.ts

```diff
@@ -176,7 +193,65 @@ export default function multilanguage(userOptions: Options) {
           document.head.appendChild(meta);
           document.head.appendChild(document.createTextNode("\n"));
         }
+
+        if (page.data.unmatchedLangUrl) {
+          appendHreflang("x-default", page.data.unmatchedLangUrl, document);
+        }
       }
     });
   };
 }
+
+function getUnmatchedLangPath(
+  currentPage: Page<Data>,
+  filteredPages: Page<Data>[],
+): string | undefined {
+  const { unmatchedLangUrl } = currentPage.data;
+
+  if (!unmatchedLangUrl) return void 0;
+
+  // If unmatchedLang is an external URL string
+  if (unmatchedLangUrl.startsWith("http")) {
```

Comment:
This condition is a bit weak. Maybe `if (URL.canParse(unmatchedLangUrl))` is better because it checks if the value is a valid URL.